### PR TITLE
feat: allow private packages to be versioned

### PR DESCRIPTION
**Pull request by essarn**
PR for feature request: #44 

Private packages can now be versioned by including the `--all [-a]` flag when running the `version` command.
Also added documentation for this new flag.

### CODE REVIEW

File: packages/melos/lib/src/common/git.dart [hunk 1]
```diff
@@ -137,9 +137,13 @@ Future<void> gitCommit(String message, {String workingDirectory}) async {
...
-  if (package.isPrivate) {
+Future<List<GitCommit>> gitCommitsForPackage(
+  MelosPackage package, {
+  String since,
+  String preid = 'dev',
+  bool versionAll,
+}) async {
+  if (!versionAll && package.isPrivate) {
     return [];
   }
```
Comment (Salakar) [inline packages/melos/lib/src/common/git.dart:148]: I think it'd be safe to remove the private checking logic here and the new entirely as the only 2 calls that use it already do either private/no-private filters or `all` filters themselves

File: packages/melos/lib/src/command/version.dart [hunk 2]
```diff
@@ -365,15 +373,19 @@ class VersionCommand extends Command {
...
+          .toList();
+    }
 
     if (pendingPackageUpdates.isEmpty) {
       logger.stdout(AnsiStyles.yellow(
           'No packages were found that required versioning.'));
       logger.stdout(AnsiStyles.gray(
           'Hint: try running "melos list" with the same filtering options to see a list of packages that were included.'));
+      logger.stdout(AnsiStyles.gray(
+          'Hint: try running "melos version --all" to include private packages'));
```
Comment (Salakar) [inline packages/melos/lib/src/command/version.dart:388]:
```suggestion
          'Hint: try running "melos version --all" to include private packages.'));
```

**Comment by Salakar** [comment]
Nice, thanks so much for sending this. 2 very minor points but after them this looks good to me

File: packages/melos/lib/src/common/git.dart (hunk 1 above)
Comment (essarn) [reply packages/melos/lib/src/common/git.dart]: Yes you are right, removed the check.

**Review by Salakar — APPROVED** [review]

**Comment by essarn** [comment]
No problem, happy to help. 
Currently doing a refactor of a project with `melos` and really enjoying the package.

**Comment by Salakar** [comment]
> Currently doing a refactor of a project with melos and really enjoying the package.

Good to hear :D too many late nights went into it so I'm glad it's being used

**Comment by Salakar** [comment]
`0.4.7` has been published with this in, though you now have a local copy setup now which could be good to keep for potential future contributions 🤓

**Comment by essarn** [comment]
> `0.4.7` has been published with this in, though you now have a local copy setup now which could be good to keep for potential future contributions 🤓

Definitely! An interesting thing to do in the future could be to extend the package to version and publish single package projects (not a monorepo) like [cider](https://pub.dev/packages/cider) does.

**Comment by Salakar** [comment]
Oh interesting, not seen cider before, interesting approach. I can see the change log commands being useful or you could also in theory run cider commands themselves on each package via `melos exec -- coder ...`

Also being able to specify a version bump type (as an alternative to specifying exact version), e.g.  `melos version <packageName> <breaking|major|minor|patch|build|packageVersion>` would be cool to support on melos - been on my todo for a while just not got around to it

**Comment by essarn** [comment]
> Oh interesting, not seen cider before, interesting approach. I can see the change log commands being useful or you could also in theory run cider commands themselves on each package via `melos exec -- coder ...`
> 
> Also being able to specify a version bump type (as an alternative to specifying exact version), e.g. `melos version <packageName> <breaking|major|minor|patch|build|packageVersion>` would be cool to support on melos - been on my todo for a while just not got around to it

The version bump feature would for sure be nice to support 👀 

Not a fan of the log command/system though, prefer to just base changelog generation and versioning straight off conventional commits.

**Pull request merged**